### ISSUE TITLE
OLGoogleMaps option to disable Tile, Image, and ImageWMS watchlayers

### ISSUE THREAD

**Issue by sindrma** [mapgears/ol3-google-maps#94]
Hi!
This is just a suggestion in case there are others out there wanting something similar. 

The application my firm is working on uses Clusters and styles currently not supported by ol3-google-maps.  We, therefore, decided to disable watchlayers for Tiles, Images, and ImageWMS as we want to keep OL3 layers. Of course, this will make the OL3 layers lag during zoom and when moving around in the map. However, we are able to keep all styling we have defined without the need for making big changes in our ol3-google-maps fork.

We defined a watchFeature object which is an optional parameter passed to olgm.OLGoogleMaps:

``` javascript
var watchFeatures = options.watchFeatures !== undefined ?
      options.watchFeatures : {image: true, imageWMS: true, tile: true};
```

It is further passed down to olgm.layers.Herald, and used to disable the different watchlayers as below:

``` javascript
olgm.herald.Layers.prototype.watchLayer_ = function(layer) {
  if (layer instanceof olgm.layer.Google) {
    this.watchGoogleLayer_(layer);
  } else if (layer instanceof ol.layer.Vector && this.watchVector_) {
    this.vectorSourceHerald_.watchLayer(layer);
  } else if (layer instanceof ol.layer.Tile && this.watchFeatures_.tile) {
    this.tileSourceHerald_.watchLayer(layer);
  } else if (layer instanceof ol.layer.Image && this.watchFeatures_.image) {
    var source = layer.getSource();
    if (source instanceof ol.source.ImageWMS && this.watchFeatures_.imageWMS) {
      this.imageWMSSourceHerald_.watchLayer(layer);
    }
  } 
};
```

If the option is wanted, we would gladly submit a PR on it.  

We also had to disable geometry event listeners on features in featureherald.js. This is probably due to our use of custom features.

**Comment by adube** [comment]
Sure, a PR would be nice.  If you could create an example demonstrating this use-case, it would be helpful too.

Thanks!

**Comment by sindrma** [comment]
A short example would be when we are creating the OLGoogleMaps object:

``` javascript
var olgmOptions = {map: this.map, watchVector: false,
                            watchFeatures: {image: false, imageWMS: false, tile: false}};
this.olGM = new olgm.OLGoogleMaps(olgmOptions);
this.olGM.activate();
```

This disables all Google layers except the map, and allows us to keep OL3 layers on top of the Google Map. If wanted, we could create a pull request on it. Just let me know.

As mentioned, we also had to disable event listeners in featureherald.js, or more specifically in the function herald.Feature.activate:

``` javascript
 // event listeners (todo)
// commented out in our fork
var keys = this.listenerKeys;
this.geometryChangeKey_ = geometry.on('change',
                                      this.handleGeometryChange_,
                                      this);
keys.push(this.geometryChangeKey_);
keys.push(this.feature_.on(
    'change:' + this.feature_.getGeometryName(),
    this.handleGeometryReplace_, this
    ));
```

I am not sure why it does not work for us, but I expect it is because we use custom features in our OL3 fork.

**Comment by adube** [comment]
To have the possibility to watch the type of layers that we want seems pretty cool.

`watchVector` is the option used to watch vector layers (and its features).  I think it would make sense to have 'sibling' options instead, such as `watchWMS`, `watchTile`, etc.  What do you think ?

Not sure what's going on with the features herald, but you could inspect what happens with a Dev tools debugger.  You could create a branch with your example in it so that we could test it too.

**Comment by sindrma** [comment]
I agree with you, you could use sibling options. Either that or change into using a `watch`-object which includes the on/off-option for each of the possible layers, as done in the code above. It would then also include the option for the vector layer.

If I get time, I will do some more investigation on why geometry change listeners does not work for us.

**Comment by samuellapointe** [comment]
Fixed with #111. The syntax isn't exactly the same as in your example; we send a `watch` object to the olGM object containing the following attributes:
- `image` for image layers, we only have imageWMS supported anyway
- `tile` for tiled layers and
- `vector` for vector layers, that one replaces the old watchVector option.

See [this file](https://github.com/mapgears/ol3-google-maps/blob/master/examples/watchoptions.js) for an example of the syntax.
